bmx280.c

```diff
@@ -122,7 +139,27 @@ struct bmx280_t{
  * Returns false if the sensor was not found.
  * @param bmx280 The driver structure.
  */
+#if !(CONFIG_USE_I2C_NEW_DEVICE)
 #define bmx280_validate(bmx280) (!(bmx280->slave == 0xDE && bmx280->chip_id == 0xAD))
+#else
+#define bmx280_validate(bmx280) (!(bmx280->i2c_dev == NULL && bmx280->chip_id == 0xAD))
+#endif
+
+#if CONFIG_USE_I2C_NEW_DEVICE
+/**
+ * Read from sensor.
+ * @param bmx280 Driver Sturcture.
+ * @param dev_addr Chip addresses.
+ */
+static void bmx280_device_create(bmx280_t *bmx280, const uint16_t dev_addr)
+{
+    ESP_LOGI("bmx280", "device_create for BMP280/BME280 sensors on ADDR %X", dev_addr);
+    bmx280->i2c_cfg->dev_cfg.device_address = dev_addr;
+    // Add device to the I2C bus
+    ESP_ERROR_CHECK(i2c_master_bus_add_device(bmx280->i2c_cfg->bus_handle, &bmx280->i2c_cfg->dev_cfg, &bmx280->i2c_dev));
```

Comment:
Don't assert for the user at runtime, I suggest you bubble this error up by returning the error value.